### PR TITLE
Use Ember.assign instead of Ember.merge

### PR DESCRIPTION
**Pull request by amk221**
http://emberjs.com/deprecations/v2.x/#toc_deprecations-added-in-2-5

### CODE REVIEW

File: app-shims.js [hunk 1]
```diff
@@ -133,7 +133,7 @@
         'default': Ember.Object
       },
       'ember-platform': {
-        'assign':         Ember.merge,
+        'assign':         Ember.assign,
```
Comment (rwjblue) [inline app-shims.js]: Can you make this fallback?

    Ember.assign || Ember.merge

Comment (amk221) [inline app-shims.js]: ah, sure. Done

**Comment by rwjblue** [comment]
Thank you!

**Comment by john-kurkowski** [comment]
Are `merge` and `assign` the same thing in Ember though? The linked blog post doesn't mention `assign` as a replacement for `merge`. 

They definitely behave different in Lodash.

**Pull request merged**